File: test-suite/java/security/RandomSecurityClean.java
import java.security.SecureRandom;
import java.util.Base64;
import java.util.Random;

class RandomSecurityClean {
    private static final SecureRandom SECURE_RANDOM = new SecureRandom();

    String resetToken() {
        byte[] bytes = new byte[32];
        SECURE_RANDOM.nextBytes(bytes);
        return Base64.getUrlEncoder().withoutPadding().encodeToString(bytes);
    }

    String sessionSecret() {
        return resetToken();
    }

    String csrfNonce() {
        return resetToken();
    }

    String apiKey() {
        return "ak_" + resetToken();
    }

    String pickDisplayTheme() {
        Random displayRandom = new Random(42);
        return new String[] {"light", "dark", "system"}[displayRandom.nextInt(3)];
    }
}

File: test-suite/csharp/security/RandomSecurityClean.cs
using System;
using System.Security.Cryptography;

public sealed class RandomSecurityClean
{
    public string ResetToken()
    {
        return Convert.ToHexString(RandomNumberGenerator.GetBytes(32));
    }

    public string SessionSecret()
    {
        return ResetToken();
    }

    public string CsrfNonce()
    {
        return Convert.ToBase64String(RandomNumberGenerator.GetBytes(32));
    }

    public string ApiKey()
    {
        return "ak_" + RandomNumberGenerator.GetHexString(32);
    }

    public string OneTimePassword()
    {
        return RandomNumberGenerator.GetInt32(100000, 1000000).ToString();
    }

    public string PickDisplayTheme()
    {
        var displayRandom = new Random(42);
        var themes = new[] { "light", "dark", "system" };
        return themes[displayRandom.Next(themes.Length)];
    }
}
